cmake/compiler/xcc-clang/compiler_flags.cmake

```diff
@@ -1,11 +1,18 @@
 # SPDX-License-Identifier: Apache-2.0
 
 include(${ZEPHYR_BASE}/cmake/compiler/clang/compiler_flags.cmake)
+include(${ZEPHYR_BASE}/cmake/compiler/xcc/target.cmake)
 
 # Clear "nostdinc"
 set_compiler_property(PROPERTY nostdinc)
 set_compiler_property(PROPERTY nostdinc_include)
 
+if(CONFIG_CPP)
+  set(NOSTDINC_INCLUDE ${NOSTDINC})
+  list(REMOVE_ITEM NOSTDINC_INCLUDE include-fixed)
+  set_compiler_property(PROPERTY nostdinc_include "${NOSTDINC_INCLUDE}")
+endif()
```

Comment:
I would recommend either adding a comment here, and/or expand upon the reasoning in the commit message explaining why this is needed.

Reply:
I wonder where NOSTDINC comes from and why it needs to be adjusted.

`NOSTDINC_INCLUDE` has the same name as `nostdinc_include` above, is this intended? I don't understand the relationship between the two.

Reply:
@marc-hb Updated this code to avoid confusion. NOSTDINC defined in the `target.cmake` already defined the list NOSTDINC to contain path to compiler-specific header files, so an initial attempt was to re-use that. And NOSTDINC_INCLUDE was just a copy to avoid modifying NOSTDINC itself which might have been used in other places. But I think it would be easier to understand with the updated code. Please re-review :)

Reply:
@peter-mitsis Added the commit message.